### PR TITLE
[12.0][ENH] hr_expense_petty_cash - add journal in petty cash holder

### PR DESCRIPTION
**Pull request by Saran440**
Add journal in petty cash holder.
when bill and expense is petty cash, it will default journal from petty cash holder.

### CODE REVIEW

File: hr_expense_petty_cash/models/account_invoice.py [hunk 1]
```diff
@@ -74,3 +75,6 @@ def _onchange_is_petty_cash(self):
                 raise ValidationError(_('%s is not a petty cash holder') %
                                       self.partner_id.name)
             self._add_petty_cash_invoice_line(petty_cash)
+            if petty_cash.journal_id:
+                ctx = {'default_journal_id': petty_cash.journal_id.id}
```
Comment (kittiu) [inline hr_expense_petty_cash/models/account_invoice.py]: You want to overwite the ctx? not ctx.update({...}) ?

File: hr_expense_petty_cash/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -3,6 +3,6 @@ This module work about expense that paid by petty cash.
 General Process:
 
 #. Create Petty Cash Holder
-#. Transfer cash to Petty Cash Holder By Vendor bill
+#. Transfer cash to Petty Cash Holder
```
Comment (kittiu) [inline hr_expense_petty_cash/readme/DESCRIPTION.rst]: I think this is good for user to know how to add more cash.
```suggestion
#. Transfer cash to Petty Cash Holder (by using Vendor Bill)
```

File: hr_expense_petty_cash/models/hr_expense.py [hunk 3]
```diff
@@ -45,3 +45,11 @@ def _get_account_move_line_values(self):
             line['partner_id'] = expense.petty_cash_id.partner_id.id
             res[expense.id][1] = line
         return res
+
+    @api.multi
+    def _prepare_move_values(self):
+        self.ensure_one()
+        move_values = super()._prepare_move_values()
+        if self.payment_mode == 'petty_cash' and self.petty_cash_id.journal_id:
```
Comment (kittiu) [inline hr_expense_petty_cash/models/hr_expense.py]: I think we should not change the journal like this. It should be using the sheet.jounal_id to be correct.
But instead, when sheet is created, the petty cash's journal_id should be assigned.

**Review by kittiu — COMMENTED** [review]
Thank you, just a few points to correct.
Also, please add some more test.

**Comment by Saran440** [comment]
@kittiu Fixed. Thank you for your advise.

- change default  journal_id in hr.expense.sheet, if you're select journal in petty.cash.

**Review by kittiu — APPROVED** [review]

**Review by ps-tubtim — APPROVED** [review]
LGTM

**Review by marcelsavegnago — APPROVED** [review]

**Pull request merged**